openstates/in/bills.py

```diff
@@ -77,9 +81,127 @@ def get_name(self,random_json):
+                elif currently_counting == "":
+                    pass
+                elif re.search(r'v\. \d\.\d',l):
+                    #this gets rid of the version number
+                    #which is often found at the bottom of the doc
+                    pass
+                else:
+                    voters = l.split("  ")
+                    for v in voters:
+                        if v.strip():
+                            vote[currently_counting].append(v.strip())
+
+            assert (len(vote["yes_votes"]) == vote["yes_count"],
+                "Yes vote counts ({count}) don't match count of actual votes ({actual})".format(count=vote["yes_count"],actual=len(vote["yes_votes"])))
+            assert (len(vote["no_votes"]) == vote["no_count"],
+                "No vote counts ({count}) don't match count of actual votes ({actual})".format(count=vote["no_count"],actual=len(vote["no_votes"])))
+            assert (len(vote["other_votes"]) == vote["other_count"],
+                "Other vote counts ({count}) don't match count of actual votes ({actual})".format(count=vote["other_count"],actual=len(vote["other_votes"])))
+            
+
+            #vote should have the majority if it passed
+            if vote["passed"] and vote["yes_count"] <= vote["no_count"]:
+                raise AssertionError("The vote should have at least a majority if it passed.")
+
+            #if vote has the majority and didn't pass, throw a warning
+            #this is possible in supermajority cases,
+            #but we might want to think about it if it starts happening lots
+            if not vote["passed"] and vote["yes_count"] > vote["no_count"]:
+                self.logger.warning("Vote got a majority but didn't pass. Probably worth a look.")
+
+
+            bill.add_vote(vote)
+
+    def deal_with_version(self,version,bill,proxy):
+        #documents
+        docs = {"Amendment":version["amendments"],
+                "Fiscal Note":version["fiscal-notes"],
+                "Committee Amendment":version["cmte_amendments"],
+                "Floor Amendment":version["floor_amendments"],
+                "Committee Report":version["committee-reports"]
+                }
+        for doc_type,doc_list in docs.items():
+            for doc in doc_list:
+                title = "{doc_type}: {name}".format(doc_type=doc_type,name=doc["name"])
+                url = proxy["url"] + doc["link"]
+                bill.add_document(title,url,mimetype="application/pdf")
+
+        #version
+        link = proxy["url"] + version["link"]
+        name = version["stageVerbose"]
+        bill.add_version(name,link,mimetype="application/pdf")
+
+        #votes
+        votes = version["rollcalls"]
+        self.process_votes(votes,bill,proxy)
+
+
     def scrape(self, session, chambers):
         
         api_base_url = "https://api.iga.in.gov"
+        proxy = {"url":"http://in.proxy.openstates.org"}
```

Comment:
Maybe add a comment about the proxy, where its code is stored, and why it's needed? Seems somewhat opaque to outsiders at the moment.